File: object_Movie/src/main/java/objectstudy/condition/refund/TimeCondition.java
package objectstudy.condition.refund;


import objectstudy.Refund;
import objectstudy.Reservation;
import objectstudy.Screening;

import java.time.LocalDateTime;

public class TimeCondition implements RefundCondition {

    private LocalDateTime currentTime;

    public TimeCondition(LocalDateTime currentTime) {
        this.currentTime = currentTime;
    }

    @Override
    public boolean isSatisfiedBy(Reservation reservation) {
        return currentTime.compareTo(reservation.getScreeningTime()) <= 0;
    }
}

File: object/object/src/java_to_kotlin/condition/refund/TimeCondition.kt
package java_to_kotlin.condition.refund

import java_to_kotlin.Screening
import java.time.LocalDateTime

class TimeCondition : RefundCondition {
    private val currentTime: LocalDateTime? = null
    override fun isSatisfiedBy(screening: Screening): Boolean {
        return currentTime!!.compareTo(screening.startTime) <= 0
    }
}
